internal/venice-test-common/src/main/java/com/linkedin/venice/utils/TestUtils.java

```diff
@@ -897,4 +899,50 @@ public static void checkMissingFieldInAvroRecord(GenericRecord record, String fi
       assertEquals(e.getMessage(), "Not a valid schema field: " + fieldName);
     }
   }
+
+  public static List<String> findFoldersWithFileExtension(File directory, String fileExtension) {
+    List<String> result = new ArrayList<>();
+
+    if (!directory.exists() || !directory.isDirectory()) {
+      return result;
+    }
+    return searchForFileExtension(directory, fileExtension.toLowerCase());
+  }
+
+  public static List<String> searchForFileExtension(File directory, String fileExtension) {
+    List<String> result = new ArrayList<>();
+    if (!directory.canRead()) {
+      System.err.println("Cannot read directory: " + directory.getAbsolutePath());
+      return result;
+    }
+    File[] files = directory.listFiles();
+    if (files == null) {
+      System.err.println("Error reading directory: " + directory.getAbsolutePath());
+      return result;
+    }
+    for (File file: files) {
+      if (file.isDirectory() && !file.isHidden()) {
+        result.addAll(searchForFileExtension(file, fileExtension)); // Recursively search subdirectories
+      } else if (file.isFile() && !file.isHidden() && file.getName().toLowerCase().endsWith(fileExtension)) {
+        result.add(file.getParent()); // Add the parent directory to the result list
+        break; // Stop searching this directory once we find a file with the desired extension
+      }
+    }
+    return result;
+  }
+
+  public static boolean directoryContainsFolder(String directoryPath, String folderName) {
+    File directory = new File(directoryPath);
+    if (directory.exists() && directory.isDirectory()) {
```

Comment:
AFIAK, `isDirectory` checks whether it exists or not, the `directory.exists()` is redundant. We can remove it